modules/custom/az_paragraphs/src/Plugin/paragraphs/Behavior/AZDefaultParagraphBehavior.php

```diff
@@ -0,0 +1,154 @@
+<?php
+
+namespace Drupal\az_paragraphs\Plugin\paragraphs\Behavior;
+
+use Drupal\Core\Entity\Display\EntityViewDisplayInterface;
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\paragraphs\Entity\Paragraph;
+use Drupal\paragraphs\ParagraphInterface;
+use Drupal\paragraphs\ParagraphsBehaviorBase;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+
+/**
+ * Provides a base behavior for Quickstart Paragraph Defaults.
+ *
+ * @ParagraphsBehavior(
+ *   id = "az_default_paragraph_behavior",
+ *   label = @Translation("Default Quickstart Paragraph Behavior"),
+ *   description = @Translation("Provides default Quickstart paragraph settings."),
+ *   weight = 0
+ * )
+ */
+class AZDefaultParagraphBehavior extends ParagraphsBehaviorBase {
+
+  /**
+   * Drupal\Core\Entity\EntityDisplayRepositoryInterface definition.
+   *
+   * @var \Drupal\Core\Entity\EntityDisplayRepositoryInterface
+   */
+  protected $entityDisplayRepository;
+
+  /**
+   * Drupal\Core\Entity\EntityTypeManagerInterface definition.
+   *
+   * @var \Drupal\Core\Entity\EntityTypeManagerInterface
+   */
+  protected $entityTypeManager;
+
+  /**
+   * Drupal\Core\Asset\LibraryDiscoveryInterface definition.
+   *
+   * @var \Drupal\Core\Asset\LibraryDiscoveryInterface
+   */
+  protected $libraryDiscovery;
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container, array $configuration, $plugin_id, $plugin_definition) {
```

Comment:
I think the preferred way of injecting additional dependencies (beyond what the parent class does is to actually override the `__construct()` method with additional typeHinted dependencies in the method signature, and then inject all of the dependencies (both those defined by the parent and the additional ones defined in the extended class) in the `create()` method.  We have an example of this in the az_quickstart_core module here:
https://github.com/az-digital/az_quickstart/blob/main/modules/custom/az_core/src/Form/QuickstartCoreSettingsForm.php#L31-L57

Reply:
@tadean This documentation on drupal.org provides some similar examples: https://www.drupal.org/docs/8/api/services-and-dependency-injection/dependency-injection-for-a-form

Reply:
Thanks @joeparsons ! Updated to the style of overriding that overrides the constructor.